File: 프로그래머스/LV2/코딩테스트 연습/2018 KAKAO BLIND RECRUITMENT/[3차] n진수 게임/[3차] n진수 게임.cpp
#include <string>
#include <vector>

using namespace std;

string to_n(int i, int n) {
    string s = "";
    char r;

    while (i) {
        r = i % n;
        i = i / n;
        if(i >= 10 && i <= 15)s += (char)(r+55);
        else s += (char)r;
    }

    reverse(s.begin(), s.end());
    return s;
}

string solution(int n, int t, int m, int p) {
    string answer = "";
    string s = "0";

    int i = 1;
    while (s.length() < t*m) {
        s += to_n(i,n);
        i++;
    }
    
    i = p-1;
    while (t) {
        answer += s[i];
        i += m;
        t--;
    }
    
    return answer;
}

File: 프로그래머스/LV2/코딩테스트 연습/2018 KAKAO BLIND RECRUITMENT/[3차] n진수 게임/[3차] n진수 게임.py
def to_n(i, n):
    s = ''
    while i:
        i, r = divmod(i, n)
        if 10 <= r <= 15:
            s += chr(r+55)
        else:
            s += str(r)
    return s[::-1]


def solution(n, t, m, p):
    answer = ""
    s = "0"

    i = 1
    while len(s) < t*m:
        s += to_n(i, n)
        i += 1
    
    i = p-1
    while t:
        answer += s[i]
        i += m
        t -= 1
    return answer